From 754e093920a80deb1f9306e3be26dd7e35ba8ef8 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 26 Mar 2014 15:13:09 +0800
Subject: [PATCH 505/635] ENGR00304148 - Picture color is wrong for 1M
 pixel/VGA/QVGA with ov5642 sensor

Partly fix this issue.
ov5642 is not work good on 15fps, change range from [25, 30] to [23, 30] to make
framework choose [23,30] fps range, so 30 fps.
For ov5642, seems different exposure cause color, brightness unstable, need driver to fix.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera2/Ov5640Csi.cpp | 4 ++--
 mx6/libcamera2/Ov5642Csi.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/mx6/libcamera2/Ov5640Csi.cpp b/mx6/libcamera2/Ov5640Csi.cpp
index 29547ce..1fed6f1 100755
--- a/mx6/libcamera2/Ov5640Csi.cpp
+++ b/mx6/libcamera2/Ov5640Csi.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -139,7 +139,7 @@ status_t Ov5640Csi::initSensorInfo(const CameraInfo& info)
     i = 0;
     mTargetFpsRange[i++] = 10;
     mTargetFpsRange[i++] = 15;
-    mTargetFpsRange[i++] = 25;
+    mTargetFpsRange[i++] = 23;
     mTargetFpsRange[i++] = 30;
 
     setMaxPictureResolutions();
diff --git a/mx6/libcamera2/Ov5642Csi.cpp b/mx6/libcamera2/Ov5642Csi.cpp
index 69ea429..bfa1ee6 100755
--- a/mx6/libcamera2/Ov5642Csi.cpp
+++ b/mx6/libcamera2/Ov5642Csi.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -132,7 +132,7 @@ status_t Ov5642Csi::initSensorInfo(const CameraInfo& info)
     i = 0;
     mTargetFpsRange[i++] = 10;
     mTargetFpsRange[i++] = 15;
-    mTargetFpsRange[i++] = 25;
+    mTargetFpsRange[i++] = 23;
     mTargetFpsRange[i++] = 30;
 
     setMaxPictureResolutions();
-- 
1.8.0

